rustls/src/crypto/ring/ticketer.rs

```diff
@@ -100,10 +106,23 @@ impl ProducesTickets for AeadTicketer {
                 ciphertext
             })
             .ok()
+            .map(|r| {
+                self.maximum_ciphertext_len
+                    .fetch_max(r.len(), Ordering::SeqCst);
+                r
+            })
     }
 
     /// Decrypt `ciphertext` and recover the original message.
     fn decrypt(&self, ciphertext: &[u8]) -> Option<Vec<u8>> {
+        if ciphertext.len()
+            > self
+                .maximum_ciphertext_len
+                .load(Ordering::SeqCst)
+        {
+            return None;
```

Comment:
Does this case merit some kind of a warning log? It seems like it shouldn't occur under normal circumstances.

Reply:
I think we should avoid doing any extraneous work in this branch (or any other ticket rejection branches). However I'd be open to doing this gated by `#[cfg(debug_assertions)]`?

Reply:
Sounds reasonable 👍